api/receiver/appengine.go

```diff
@@ -57,12 +58,21 @@ func NewAppEngineAPI(ctx context.Context) AppEngineAPI {
 	return &appEngineAPIImpl{
 		AppEngineAPIImpl: shared.NewAppEngineAPI(ctx),
 		queue:            ResultsQueue,
+		store:            shared.NewAppEngineDatastore(ctx, false),
 	}
 }
 
 func (a *appEngineAPIImpl) addTestRun(testRun *shared.TestRun) (*DatastoreKey, error) {
-	key := datastore.NewIncompleteKey(a.Context(), "TestRun", nil)
-	key, err := datastore.Put(a.Context(), key, testRun)
+	ctx := a.Context()
+	var key shared.Key
+	var err error
+	store := shared.NewAppEngineDatastore(ctx, false)
+	key = store.NewIDKey("TestRun", testRun.ID)
```

Comment:
`tesRun.ID` needs to be manually set here:
https://github.com/web-platform-tests/wpt.fyi/blob/03bc0e68e21cd92b57fad345acd5475d873968d5/api/receiver/create_run.go#L63

Reply:
No it doesn't - the `id` field in the JSON either is or is not set by the POSTer.

https://github.com/web-platform-tests/wpt.fyi/pull/1168/files#diff-696fea2c57fe75ba995a921eb5585dddR535

Reply:
Ahh I didn't see that `TestRun.ID` also had a `json` tag, so this would just work automatically. Great.